lib/asynchronousSender.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache License. See License in the project root for license information.
+
+import {
+   Delivery, Message, Sender as RheaSender, SessionEvents
+} from "rhea";
+import * as log from "./log";
+import { BaseSender, BaseSenderOptions } from "./sender";
+import { SenderEvents } from "rhea";
+import { OnAmqpEvent, EventContext } from "./eventContext";
+import { Session } from "./session";
+import { OperationTimeoutError } from "./operationTimeoutError";
+
+/**
+ * @internal
+ */
+export interface PromiseLike {
+  resolve: (value?: any) => void;
+  reject: (reason?: any) => void;
+  timer: NodeJS.Timer;
+}
+
+/**
+ * Describes the event listeners that can be added to the AsynchronousSender.
+ * @interface Sender
+ */
+export declare interface AsynchronousSender {
```

Comment:
Maybe for brevity's sake use `Async` instead of `Asynchronous`? It helps by the time we reach `AsynchronousSenderWithSessionOptions` :)

Reply:
True. Will name it `AsyncSender`